poseidon2-circuit-air/src/air.rs

```diff
@@ -0,0 +1,680 @@
+            .from_env_lossy();
+
+        Registry::default()
+            .with(env_filter)
+            .with(ForestLayer::default())
+            .init();
+    }
+
+    #[test]
+    fn prove_poseidon2_sponge() -> Result<
+        (),
+        p3_uni_stark::VerificationError<
+            p3_fri::verifier::FriError<
+                p3_merkle_tree::MerkleTreeError,
+                p3_merkle_tree::MerkleTreeError,
+            >,
+        >,
+    > {
+        init_logger();
+        type Val = BabyBear;
+        type Challenge = BinomialExtensionField<Val, 4>;
+
+        type ByteHash = Keccak256Hash;
+        let byte_hash = ByteHash {};
+
+        type U64Hash = PaddingFreeSponge<KeccakF, 25, 17, 4>;
+        let u64_hash = U64Hash::new(KeccakF {});
+
+        type FieldHash = SerializingHasher<U64Hash>;
+        let field_hash = FieldHash::new(u64_hash);
+
+        type MyCompress = CompressionFunctionFromHasher<U64Hash, 2, 4>;
+        let compress = MyCompress::new(u64_hash);
+
+        // WARNING: DO NOT USE SmallRng in proper applications! Use a real PRNG instead!
+        type ValMmcs = MerkleTreeHidingMmcs<
+            [Val; p3_keccak::VECTOR_LEN],
+            [u64; p3_keccak::VECTOR_LEN],
+            FieldHash,
+            MyCompress,
+            SmallRng,
+            4,
+            4,
+        >;
+        let mut rng = SmallRng::seed_from_u64(1);
+        let val_mmcs = ValMmcs::new(field_hash, compress, rng.clone());
+
+        type ChallengeMmcs = ExtensionMmcs<Val, Challenge, ValMmcs>;
+        let challenge_mmcs = ChallengeMmcs::new(val_mmcs.clone());
+
+        type Challenger = SerializingChallenger32<Val, HashChallenger<u8, ByteHash, 32>>;
+        let challenger = Challenger::from_hasher(vec![], byte_hash);
+
+        let fri_params = create_benchmark_fri_params(challenge_mmcs);
+
+        let beginning_full_constants = rng.random();
+        let partial_constants = rng.random();
+        let ending_full_constants = rng.random();
+
+        let constants = RoundConstants::new(
+            beginning_full_constants,
+            partial_constants,
+            ending_full_constants,
+        );
+
+        let perm = Poseidon2::<
+            Val,
+            Poseidon2ExternalLayerBabyBear<WIDTH>,
+            Poseidon2InternalLayerBabyBear<WIDTH>,
+            WIDTH,
+            SBOX_DEGREE,
+        >::new(
+            ExternalLayerConstants::new(
+                beginning_full_constants.to_vec(),
+                ending_full_constants.to_vec(),
+            ),
+            partial_constants.to_vec(),
+        );
+
+        let air: Poseidon2CircuitAir<
+            Val,
+            GenericPoseidon2LinearLayersBabyBear,
+            D,
+            WIDTH,
+            WIDTH_EXT,
+            RATE_EXT,
+            CAPACITY_EXT,
+            SBOX_DEGREE,
+            SBOX_REGISTERS,
+            HALF_FULL_ROUNDS,
+            PARTIAL_ROUNDS,
+        > = Poseidon2CircuitAir::new(constants.clone());
```

Comment:
similarly to the above suggestion, would be nice if you were defining some types for commonly used configurations to make it lighter for users, maybe in some `public_types.rs` module or similar